src/autocomplete.js

```diff
@@ -0,0 +1,193 @@
+const monthDay = `\\d+ +${ext}`
+const monthDays = exts.map(
+  extValue => `\\d+ ${extValue}`
+)
+
+const byMonthDay = `the ${monthDay} (${and} +${monthDay})*`
+const byMonthDays = monthDays.flatMap(
+  monthDayValue => [`in ${monthDayValue}`, ...ands.flatMap(
+    andValue => [`in ${monthDayValue}${andValue}`, ...monthDays.map(
+      subMonthDayValue => `in ${monthDayValue}${andValue} ${subMonthDayValue}`
+    )]
+  )]
+)
+
+const yearDay = `\\d+ +${ext}`
+const yearDays = exts.map(
+  extValue => `\\d+ ${extValue}`
+)
+
+const byYearDay = `on +the +${yearDay} +${ext} +(${and} +${yearDay} +${ext})* +day +of +the +year`
+const byYearDays = yearDays.flatMap(
+  yearDayValue => [`on the ${yearDayValue}`, ...exts.flatMap(
+    extValue => [`on the ${yearDayValue} ${extValue}`, ...ands.flatMap(
+      andValue => [`on the ${yearDayValue} ${extValue}${andValue}`, ...yearDays.map(
+        subYearDayValue => `on the ${yearDayValue} ${extValue}${andValue} ${subYearDayValue}`
+      )]
+    )]
+  )]
+)
+
+const weekNo = '\\d+'
+
+const byWeekNo = `on +the +${weekNo} +${ext} +(${and} +${weekNo} +${ext})* +week +of +the +year`
+const byWeekNos = exts.flatMap(
+  extValue => [`on the ${weekNo} ${extValue} week of the year`, ...ands.flatMap(
+    andValue => [`on the ${weekNo} ${extValue}${andValue}`, ...yearDays.map(
+      yearDayValue => `on the ${weekNo} ${extValue}${andValue} ${yearDayValue} ${extValue} week of the year`
+    )]
+  )]
+)
+
+const hour = '[0-3]?\\dh'
+const minute = '[0-5]\\dm'
+const second = '[0-5]\\ds'
+const time = `(${hour})?(${minute})?(${second})?`
+const times = [
+  '\\d+h',
+  '\\d+h\\d+m',
+  '\\d+h\\d+m\\d+s',
+  '\\d+m',
+  '\\d+m\\d+s',
+  '\\d+s'
+]
+
+const byTime = `at +${time} +(${and} +${time})*`
+const byTimes = times.flatMap(
+  timeValue => [`at ${timeValue}`, ...ands.flatMap(
+    andValue => times.flatMap(
+      subTimeValue => [`at ${timeValue}${andValue} ${subTimeValue}`]
+    )
+  )]
+)
+
+const until = `until +month +\\d+ +, +\\d{4} +(at +${time})?`
+const untils = [
+  'until month \\d+, \\d+',
+  ...times.flatMap(
+    timeValue => `until month \\d+, \\d+ at ${timeValue}`
+  )
+]
+
+const count = 'for +\\d+ +times?'
+const counts = ['for \\d+ time', 'for \\d+ times']
+
+const re = new RegExp(`^\\s*(?<freq>${freq})?(\\s+(?<weekStart>${weekStart}))?(\\s+(?<byWeekDay>${byWeekDay}))?(\\s+(?<byMonth>${byMonth}))?(\\s+(?<byMonthDay>${byMonthDay}))?(\\s+(?<byYearDay>${byYearDay}))?(\\s+(?<byWeekNo>${byWeekNo}))?(\\s+(?<byTime>${byTime}))?(\\s+(?<until>${until}))?(\\s+(?<count>${count}))?\\s*$`, 'imu')
+
+const prefixes = {
+  freq: freqs,
+  weekStart: weekStarts,
+  byWeekDay: byWeekDays,
+  byMonth: byMonths,
+  byMonthDay: byMonthDays,
+  byYearDay: byYearDays,
+  byWeekNo: byWeekNos,
+  byTime: byTimes,
+  until: untils,
+  count: counts
+}
+
+const autocomplete = exp => {
+  const result = []
+  const match = exp.match(re)
+  if (match) {
+    const { groups } = match
+    Object.entries(prefixes).forEach(
```

Comment:
:hammer: Your `forEach` mutates result. Use `filter` and `map` instead to return a value and not mutate anything.